### PR TITLE
Kind Button set to `plain` when it has children

### PR DESCRIPTION
**Pull request by ShimiSun**
#### What does this PR do?
Kind Button set to `plain` when it has children

#### Where should the reviewer start?
doc.js
#### What testing has been done on this PR?
storybook + jest
#### How should this be manually tested?
storybook
#### Any background context you want to provide?
This PR is a resolution from discussions around the kind button.
#### Do the grommet docs need to be updated?
Done.
#### Should this PR be mentioned in the release notes?
Yes.
#### Is this change backwards compatible or is it a breaking change?
backward compatible.

### CODE REVIEW

File: src/js/components/Button/__tests__/Button-kind-test.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ describe('Button kind', () => {
     expect(results).toHaveNoViolations();
   });
 
-  test('custom theme', () => {
+  test('default button', () => {
     const { container } = render(
       <Grommet theme={buttonKindTheme}>
         <Button default />
```
Comment (taysea) [inline src/js/components/Button/__tests__/Button-kind-test.js:34]:
```suggestion
        <Button default />
```
do we need to have the default prop here? shouldn't it be applied automatically by the code/would it be good to remove to ensure it's being applied automatically

**Review by taysea — COMMENTED** [review]
LGTM. just one small comment.

**Review by ericsoderberghp — COMMENTED** [review]
Just wondering if we can add `disabled` to the children function argument.

File: src/js/components/Button/README.md [hunk 2]
```diff
@@ -134,8 +134,8 @@ Function that can be called to render the visual representation.
       Button can take in Children as a function, node, or object. 
       For example hover can be passed as an object that would 
       then return a react element.
-      `children={({ hover }) => <Box...>{...}</Box>}`. When Button has
-      children, it is styled as a `plain` button.
+      `children={({ hover, focus }) => <Box...>{...}</Box>}`.
```
[the remark this answers is not in the document]
Comment (ShimiSun) [reply src/js/components/Button/README.md:137]: I'd like to add it to a separate PR with its own testing and story example.

File: src/js/components/Button/doc.js [hunk 3]
```diff
@@ -34,8 +34,8 @@ export const doc = Button => {
       Button can take in Children as a function, node, or object. 
       For example hover can be passed as an object that would 
       then return a react element.
-      \`children={({ hover }) => <Box...>{...}</Box>}\`. When Button has
-      children, it is styled as a \`plain\` button.
+      \`children={({ hover, focus }) => <Box...>{...}</Box>}\`.
```
[the remark this answers is not in the document]
Comment (ShimiSun) [reply src/js/components/Button/doc.js:37]: I'd like to add disabled on a separate PR with its own testing and story example.

**Review by ericsoderberghp — APPROVED** [review]

**Pull request merged**